tmt/steps/discover/shell.py

```diff
@@ -72,34 +115,29 @@ def go(self) -> None:
         dist_git_source = self.get('dist-git-source', False)
 
         # Check and process each defined shell test
-        assert self.data['tests'] is not None
-        for data in self.data['tests']:
+        for data in cast(DiscoverShellData, self.data).tests:
             # Create data copy (we want to keep original data for save()
             data = copy.deepcopy(data)
             # Extract name, make sure it is present
-            try:
-                name = data.pop('name')
-            except KeyError:
+            # TODO: can this ever happen? With annotations, `name: str` and `test: str`, nothing
```

Comment:
@psss what do you think? These checks are probably redundant, with schema validation and normalization, every `data` shall have `name`, `test` and `duration`.

As long as validation is not mandatory, I'd keep these around, but they shouldn't be needed after validation is applied.